src/Model/NotificationInterface.php

```diff
@@ -1,21 +1,33 @@
-<?php
-
-/*
- * This file is part of the Tabler bundle, created by Kevin Papst (www.kevinpapst.de).
- *
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace KevinPapst\TablerBundle\Model;
-
-interface NotificationInterface
-{
-    public function getIdentifier(): string;
-
-    public function getMessage(): string;
-
-    public function getType(): string;
-
-    public function getUrl(): ?string;
-}
+<?php
+
+/*
+ * This file is part of the Tabler bundle, created by Kevin Papst (www.kevinpapst.de).
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace KevinPapst\TablerBundle\Model;
+
+interface NotificationInterface

```

Comment:
I'm not sure I get it.
What is BC Breaks? The changes of return type?

Reply:
Literally any change on an interface is a BC break 🤷 Now that we are "1.x" we should limit these do a decent amount.  
The problem is: bumping composer to a new major requires every user to change composer files.
And many will not notice for a longer time that there is a new major available.

If we do not bump to a new major, we break semver and (could also break) codebases during composer update. 

What we could do to prevent that: 
- getMessage() stays as is and also returns the HTML
- type is always set, remove the ? from getType() return type
- add a second interface "NotificationNextGen" that contains the new methods, including a new one for `isHtml(): bool´

Now wrap all Notification objects in the NotificationEvent that are not `instanceof NotificationNextGen` in a NotificationModel (and trigger a deprecation). 
That should be enough as BC layer and would allow us to go forward without a new major version.

Reply:
Can you take a look at https://github.com/kevinpapst/TablerBundle/pull/176/commits/bcff26e7c2c57124b465982dcba9719340c4475e please.

I think this made it messy.
Possible alternative, can we create a new model `NotificationHtmlModel` that can only contain HTML (string),
and remove all the notification logic (active, disabled, withBadge, badgeAnimated) from there?
In the end, `NotificationEvent->notifications` can handle `NotificationModel` AND `NotificationHtmlModel`.
And it avoids the BC break more:
- NotificationModel now only has new fields (active, disabled,..) without changing the olds one.
- NotificationHtmlModel was unknown before.